File: un6unw/Week33/크레인 인형 뽑기 게임.java
import java.util.*;

public class Solution {
    public int solution(int[][] board, int[] moves) {
        Stack<Integer> basket = new Stack<>();
        int removedCount = 0;

        for (int move : moves) {
            int column = move - 1; // 배열은 0부터 시작하므로 -1

            for (int row = 0; row < board.length; row++) {
                int doll = board[row][column];

                if (doll != 0) { // 인형이 있다면
                    board[row][column] = 0; // 인형 뽑기 (0으로 비우기)

                    // 바구니가 비어있지 않고, 바구니 맨 위 인형과 같다면 제거
                    if (!basket.isEmpty() && basket.peek() == doll) {
                        basket.pop();
                        removedCount += 2;
                    } else {
                        basket.push(doll); // 그렇지 않으면 바구니에 담기
                    }
                    break; // 한 번만 뽑고 그 다음 move로 넘어감
                }
            }
        }

        return removedCount;
    }
}

File: ity0526/ity0526_Week33/크레인 인형뽑기 게임.py
def solution(board, moves):
    basket = []
    removed = 0

    for move in moves:
        col = move - 1

        # 해당 열에서 인형 찾기
        for row in range(len(board)):
            if board[row][col] != 0:
                doll = board[row][col]
                board[row][col] = 0

                if basket and basket[-1] == doll:
                    basket.pop()
                    removed += 2
                else:
                    basket.append(doll)
                break

    return removed